tiledb/sm/filter/filter.h

```diff
@@ -241,6 +258,9 @@ class Filter {
    * @param buff The buffer to serialize the data into.
    */
   virtual void serialize_impl(Serializer& serializer) const;
+
+  /** Dumps the filter details in ASCII format in the selected output string. */
+  virtual void output(std::string* out) const = 0;
```

Comment:
No. We are not doing this in the style of C, with an output argument. We have a perfectly good stream reference available; we need to use it.
```suggestion
  virtual void output(std::ostream& os) const = 0;
```